src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Serializer.cs

```diff
@@ -82,6 +88,200 @@ internal static bool CanCreateValidUri(string uriString, UriKind uriKind)
             return Uri.TryCreate(uriString, uriKind, out Uri tempUri);
         }
 
+        private Saml2EncryptedAssertion EncryptAssertion(byte[] assertionData, EncryptingCredentials encryptingCredentials)
+        {
+            ValidateEncryptingCredentials(encryptingCredentials);
+
+            var encryptedAssertion = new Saml2EncryptedAssertion();
+
+            // SymmetricSecurityKey is provided:
+            // Pre-shared symmetric key (session key) is used to encrypt an assertion
+            // Session key will not be serialized, but KeyName will be set if available
+            if (encryptingCredentials.Key is SymmetricSecurityKey)
+            {
+                encryptedAssertion.EncryptedData = CreateEncryptedDataHelper((SymmetricSecurityKey)encryptingCredentials.Key, encryptingCredentials.Enc, assertionData, encryptingCredentials);
+                encryptedAssertion.EncryptedData.KeyInfo.KeyName = encryptingCredentials.Key.KeyId;
+            }
+            // AsymmetricSecurityKey is provided:
+            // New session key will be created to encrypt an assertion
+            // Session key will be wrapped with provided AsymmetricSecurityKey
+            else if (encryptingCredentials.Key is AsymmetricSecurityKey)
+            {
+                SymmetricSecurityKey sessionKey = CreateSessionKeyBytes(encryptingCredentials.Enc);
+
+                // Create EncryptedData
+                encryptedAssertion.EncryptedData = CreateEncryptedDataHelper(sessionKey, encryptingCredentials.Enc, assertionData, encryptingCredentials);
+                // Set EncryptedData RetrievalMethodUri to ID of EncryptedKey
+                encryptedAssertion.EncryptedData.KeyInfo.RetrievalMethodUri = encryptingCredentials.Key.KeyId;
+
+                // Create EncryptedKey
+                encryptedAssertion.EncryptedKey = CreateEncryptedKeyHelper((AsymmetricSecurityKey)encryptingCredentials.Key, sessionKey, encryptingCredentials.Alg, encryptingCredentials);
+                // Add reference from EncrypedKey to EncryptedData
+                encryptedAssertion.EncryptedKey.AddReference(new DataReference(encryptedAssertion.EncryptedData.Id));
+
+                // Set Digest method for EncryptedKey
+                if (SecurityAlgorithms.RsaOaepMgf1pKeyWrap.Equals(encryptingCredentials.Alg, StringComparison.Ordinal))
+                    encryptedAssertion.EncryptedKey.EncryptionMethod.DigestMethod = SecurityAlgorithms.Sha1Digest;
```

Comment:
will thinks just work if the digest method is not set? otherwise perhaps you will have to throw an exception?

Reply:
Other algorithms may not have DigestMethod (e.g. rsa-1_5) under EncryptionMethod element. For the current implementation we are using default rsa-oaep-mgf1p parameters.